Sources/WWCalendarTimeSelector.swift

```diff
@@ -50,14 +54,26 @@ import UIKit
             showDateMonth = false
             showMonth = false
             showYear = false
+            showPicker = false
+        }
+    }
+    
+    public func showPicker(_ show: Bool) {
+        showPicker = show
+        if show && isSingular {
+            showDateMonth = false
+            showMonth = false
+            showYear = false
+            showTime = false
         }
     }
     
     fileprivate func countComponents() -> Int {
         return (showDateMonth ? 1 : 0) +
             (showMonth ? 1 : 0) +
             (showYear ? 1 : 0) +
-            (showTime ? 1 : 0)
+            (showTime ? 1 : 0) +
+            (showPicker ? 1 : 0)
     }
```

Comment:
suggestion:
```
[showDateMonth,
showMonth,
showYear,
showTime,
showPicker
].map{ $0 ? 1 : 0 }.sum
```

Reply:
no sum function in Swift, but I exchanged it with .reduce(0, +)